core/src/main/java/com/radixdlt/p2p/addressbook/AddressBook.java

```diff
@@ -158,49 +166,46 @@ public AddressBook(
     this.peerEventDispatcher = Objects.requireNonNull(peerEventDispatcher);
     this.persistence = Objects.requireNonNull(persistence);
     persistence.getAllEntries().forEach(e -> knownPeers.put(e.getNodeId(), e));
-    cleanup();
-  }
-
-  // filters out the addresses with a different network ID that have been persisted before the
-  // filtering was added
-  private void cleanup() {
-    final var cleanedUpEntries = new ImmutableMap.Builder<NodeId, AddressBookEntry>();
-    this.knownPeers.values().forEach(entry -> cleanupAddressBookEntry(entry, cleanedUpEntries));
-    this.knownPeers.clear();
-    this.knownPeers.putAll(cleanedUpEntries.build());
-  }
-
-  private void cleanupAddressBookEntry(
-      AddressBookEntry entry, ImmutableMap.Builder<NodeId, AddressBookEntry> cleanedUpEntries) {
-    final var filteredKnownAddresses =
-        entry.getKnownAddresses().stream()
-            .filter(addr -> sameNetworkHrp(addr.getUri()))
-            .collect(ImmutableSet.toImmutableSet());
+    this.numStoredAddresses =
+        knownPeers.values().stream().mapToInt(e -> e.getKnownAddresses().size()).sum();
 
-    if (filteredKnownAddresses.isEmpty() && !entry.isBanned()) {
-      // there are no known addresses and no ban info for peer so just remove it
-      this.persistence.removeEntry(entry.getNodeId());
-    } else if (filteredKnownAddresses.size() != entry.getKnownAddresses().size()) {
-      // some addresses got filtered out, need to persist a new entry
-      final var updatedEntry =
-          new AddressBookEntry(entry.getNodeId(), entry.bannedUntil(), filteredKnownAddresses);
-      cleanedUpEntries.put(entry.getNodeId(), updatedEntry);
-      persistEntry(updatedEntry);
-    } else {
-      cleanedUpEntries.put(entry.getNodeId(), entry);
+    // Reserving up to 1/10 of the address book for high priority peers
+    this.highPriorityPeers = new LRUCache<>(p2pConfig.addressBookMaxSize() / 10);
```

Comment:
I would make it `min(p2pConfig.addressBookMaxSize() / 10, MAX_NUM_VALIDATORS)` just to be sure, since it's not that big anyway.

Reply:
Hmh, from address book perspective these aren't necessarily all validators, f.e. in the latest commit I also mark the peers we're syncing with as high priority. Separately, validator set size isn't really a constant